File: Chapter06/Java/LoginDemoClient1/app/src/main/java/net/zenconsult/android/examples/LoginDemoClient1Activity.java
package net.zenconsult.android.examples;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;

import java.io.IOException;
import java.util.Arrays;

public class LoginDemoClient1Activity extends AppCompatActivity {

    private static final String TAG = "LoginDemoClient";

    private Login login;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.login).setOnClickListener(v -> {
            String u = ((EditText) findViewById(R.id.username)).getText().toString();
            String p = ((EditText) findViewById(R.id.password)).getText().toString();

            login = new Login(u, p);

            String msg = "";
            EditText text = findViewById(R.id.editText1);
            text.setText(msg);

            new Thread(() -> {
                String result = "Response is null";
                try {
                    result = login.execute();
                } catch (IOException e) {
                    Log.e(TAG, "IO Exception in reading from stream.");
                    result = "Response is null";
                }
                final String message = result;
                runOnUiThread(() -> {
                    text.setText(message);
                });
            }).start();
        });
    }
}

File: Chapter06/Kotlin/LoginDemoClient1/app/src/main/java/net/zenconsult/android/examples/LoginDemoClient1Activity.kt
package net.zenconsult.android.examples

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText

class LoginDemoClient1Activity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.login).setOnClickListener(this)
    }

    companion object {
        const val TAG = "LoginDemo1"
    }

    override fun onClick(v: View?) {
        val u = findViewById<EditText>(R.id.username).text.toString()
        val p = findViewById<EditText>(R.id.password).text.toString()

        val login = Login(u, p)

        Thread(Runnable {
            val response = login.execute()
            runOnUiThread(Runnable {
                if (response == null) {
                    findViewById<EditText>(R.id.editText1).setText("Response is null")
                } else {
                    findViewById<EditText>(R.id.editText1).setText(response)
                }
            })
        }).start()
    }
}